api/features/features_service.py

```diff
@@ -1,33 +1,109 @@
+            )
+        case _, _:
+            # For projects still fully on Core, count all overrides from Core.
+            return get_core_overrides_data(environment)
+
+
+def get_core_overrides_data(
     environment: "Environment",
-) -> typing.Dict[int, EnvironmentFeatureOverridesData]:
+    *,
+    skip_identity_overrides: bool = False,
+) -> OverridesData:
     """
     Get the number of identity / segment overrides in a given environment for each feature in the
     project.
 
     :param environment: the environment to get the overrides data for
-    :return: dictionary of {feature_id: EnvironmentFeatureOverridesData}
+    :return OverridesData: dictionary of {feature_id: EnvironmentFeatureOverridesData}
     """
     environment_feature_states_list = get_environment_flags_list(environment)
-    all_overrides_data = {}
+    all_overrides_data: OverridesData = {}
 
     for feature_state in environment_feature_states_list:
         env_feature_overrides_data = all_overrides_data.setdefault(
             feature_state.feature_id, EnvironmentFeatureOverridesData()
         )
         if feature_state.feature_segment_id:
             env_feature_overrides_data.num_segment_overrides += 1
+        elif skip_identity_overrides:
+            continue
         elif feature_state.identity_id:
             env_feature_overrides_data.add_identity_override()
-        all_overrides_data[feature_state.feature_id] = env_feature_overrides_data
+
+    return all_overrides_data
+
+
+def get_edge_overrides_data(
+    environment: "Environment",
+) -> OverridesData:
+    """
+    Get the number of identity / segment overrides in a given environment for each feature in the
+    project.
+    Retrieve identity override data from DynamoDB.
+
+    :param environment: the environment to get the overrides data for
+    :return OverridesData: dictionary of {feature_id: EnvironmentFeatureOverridesData}
+    """
+    with ThreadPoolExecutor() as executor:
+        get_environment_flags_list_future = executor.submit(
+            get_environment_flags_list,
+            environment,
+        )
+        get_overrides_data_future = executor.submit(
+            get_edge_identity_overrides,
+            environment_id=environment.id,
+        )
+    all_overrides_data: OverridesData = {}
+
+    for feature_state in get_environment_flags_list_future.result():
+        env_feature_overrides_data = all_overrides_data.setdefault(
+            feature_state.feature_id, EnvironmentFeatureOverridesData()
+        )
+        if feature_state.feature_segment_id:
+            env_feature_overrides_data.num_segment_overrides += 1
+    for identity_override in get_overrides_data_future.result():
+        env_feature_overrides_data = all_overrides_data.setdefault(
+            identity_override.feature_state.feature.id,
+            EnvironmentFeatureOverridesData(),
+        )
+        env_feature_overrides_data.add_identity_override()
```

Comment:
Technically we shouldn't have a case where we don't have the feature_id already in the dict, right?

Reply:
(I'm not saying we should necessarily change the behaviour here, but it might be worth a comment)

Reply:
Fixed.